### DISCUSSION TITLE
Spatialite?

### DISCUSSION THREAD

**Discussion by sqcsabbey** [simolus3/drift#1983] [General]
We're looking into possibly using Spatialite with our app; would that be compatible with Drift? 

By the way, [Spatialite/Flutter](https://github.com/moovida/dart_jts/issues/11) is working but not packaged.

**Comment by simolus3** [comment]
Drift is modular enough to support custom queries and SQL extensions, but I can't offer first-class support for Spatialite. If you are willing to do put in the work, you can use Spatialite with drift. I assume that you will at least need to do two things:

1. Replace `sqlite3_flutter_libs` with a custom library that bundles SpatiaLite. You can perhaps follow the build setup described [here](https://github.com/moovida/dart_jts/issues/11#issuecomment-1178780340) for that, I didn't check that though. Essentially, you want to make sure that a `libsqlite3.so` containing the SpatiaLite extension is available on Android and that SpatiaLite is statically linked into your application on iOS. 
2. Write drift utils for the spatialite functions you want to use. In drift, you can use a [`FunctionCallExpression`](https://pub.dev/documentation/drift/latest/drift/FunctionCallExpression-class.html) to write arbitrary SQL functions. You could this to write your own high-level APIs exposing SpatiaLite capabilities. They will work together with

If you have more questions on these steps, feel free to ask! I don't have the capacity to maintain decent bindings for SpatiaLite out-of-the box in drift though.
